proto/perp/v2/event.proto

```diff
@@ -12,6 +12,14 @@ option go_package = "github.com/NibiruChain/nibiru/x/perp/v2/types";
 
 // Emitted when a position changes.
 message PositionChangedEvent {
+  enum ChangeReason {
```

Comment:
Since Golang enums become integers, can we use the string enum pattern instead? It will mean you can read the name in the event:

Looks like this
```go
type DayOfWeek string

const (
    Sunday    DayOfWeek = "Sunday"
    Monday    DayOfWeek = "Monday"
    Tuesday   DayOfWeek = "Tuesday"
    Wednesday DayOfWeek = "Wednesday"
    Thursday  DayOfWeek = "Thursday"
    Friday    DayOfWeek = "Friday"
    Saturday  DayOfWeek = "Saturday"
)
```

Reply:
In our case, 
```go
type ChangeReason string

const (
    ChangeReason_OpenPosition   ChangeReason = "open_position"
    ChangeReason_ClosePosition  ChangeReason = "close_position"
    ChangeReason_AddMargin      ChangeReason = "add_margin"
    ChangeReason_RemoveMargin   ChangeReason = "remove_margin"
    ChangeReason_Liquidate      ChangeReason = "liquidate"
)
```